### PR TITLE
Add WASM response type

### PR DESCRIPTION
**Pull request by LiamGallagher737**
## Description

Added a `Wasm` response type to `axum-extra` that automatically gets the `Content-Type: application/wasm` header.

## Motivation

I was returning WASM from an API and had to implment this myself.

## Solution

Pretty much the same as the `Html` response type but with the `Content-Type: application/wasm` header.

## Additional Info

As there is no `application/wasm` in the mime crate and as the crate is no longer maintained I decided to just go with a string literal.

### CODE REVIEW

**Review by davidpdrsn — COMMENTED** [review]
Not sure this is worth having in the main axum crate. We can put it in axum-extra instead.

**Review by davidpdrsn — COMMENTED** [review]
Nice! Wanna update the changelog as well?

**Comment by LiamGallagher737** [comment]
Thank you for replying so quickly, don't think I've ever been reviewed so quick

File: axum-extra/src/response/wasm.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+/// A WASM response.
+///
+/// Will automatically get `Content-Type: application/wasm`.
+#[derive(Clone, Copy, Debug)]
+#[must_use]
+pub struct Wasm<T>(pub T);
+
+impl<T> IntoResponse for Wasm<T>
+where
+    T: Into<Full<Bytes>>,
```
Comment (jplatte) [inline axum-extra/src/response/wasm.rs:15]: Do we have the same bound in another place already? Just wondering where it comes from.

Comment (LiamGallagher737) [reply axum-extra/src/response/wasm.rs:15]: Copied from the Html IntoResponse implementation
https://github.com/tokio-rs/axum/blob/946d8c32530ae301bf26928ab22090b33366a748/axum/src/response/mod.rs#L45-L48

Comment (davidpdrsn) [reply axum-extra/src/response/wasm.rs:15]: I've actually been thinking that we could make these types work for any `T: IntoResponse`. Then it wouldn't be required to have the contents in memory.

I can that in a separate PR.

**Comment by davidpdrsn** [comment]
Superseded by https://github.com/tokio-rs/axum/pull/1921. Thanks!

**Pull request closed without merging**